lib/java/test-utils/src/main/java/org/enso/test/utils/MemoryUtils.java

```diff
@@ -59,20 +65,53 @@ private static void assertReference(
    * @param toAllocate
    * @return
    */
-  private static boolean checkAndAlloc(Reference<?> ref, ArrayList<Object> memory, int toAllocate) {
+  private static boolean checkAndAlloc(
+      PrintWriter log, Reference<?> ref, ArrayList<Object> memory, int toAllocate) {
     if (ref.get() == null) {
       return true;
     }
+    log.println("System.gc()");
     System.gc();
     if (memory != null) {
+      log.println("Allocating " + toAllocate + " bytes");
       memory.add(new byte[toAllocate]);
     }
     return false;
   }
 
-  private static void tryHarderToGc() {
+  private static void tryHarderToGc(PrintWriter log) {
+    for (var t : Thread.getAllStackTraces().entrySet()) {
+      var threadName = t.getKey().getName();
+      if (threadName.contains("TruffleCompilerThread")) {
+        var foundCompilation = false;
+        for (var frame : t.getValue()) {
+          if ("com.oracle.truffle.runtime.CompilationTask".equals(frame.getClassName())) {
+            foundCompilation = true;
+          }
+        }
+        if (foundCompilation) {
+          log.println("Found running compiler thread: " + threadName);
+          for (var frame : t.getValue()) {
+            log.println("  at " + frame);
+          }
+          try {
+            log.println("Sleeping");
+            Thread.sleep(1000);
+            log.println("Slept 1s, trying again.");
+            return;
```

Comment:
I think it would be best to have some retry mechanism to reduce chances of this error. But let's see if this change helps.